git/operations.go

```diff
@@ -41,11 +42,31 @@ func clone(ctx context.Context, workingDir string, keyRing ssh.KeyRing, repoURL,
 	return repoPath, nil
 }
 
-func commit(ctx context.Context, workingDir, commitMessage string) error {
+func commit(ctx context.Context, workingDir string, commitAction *CommitAction) error {
+	commitAuthor := commitAction.Author
+	// The user in the update.Cause.User must be a valid github user, otherwise the commit fails
+	// Hardcoding of the user to ignore is not ideal. If there is only one to ignore, ie test@test,
+	// then this is just about ok. Otherwise there could be config file with stored users to ignore etc.
+	// Currently the committer is hardcoded to flux-bot for user requested actions.
+	// TODO (potentially): allow the users to set the committer, ie have an additional
+	// update.Cause.Committer field.
+	if commitAuthor != "" && commitAuthor != "test@test" {
+		if err := config(ctx, workingDir, "flux-bot", "flux-bot@weave.works"); err != nil {
+			return errors.Wrap(err, "git commit config")
+		}
+		if err := execGitCmd(ctx,
+			workingDir, nil, nil,
```

Comment:
Might be easier just to add the author to a slice of arguments, and exec in one place.

Reply:
I thought I was adding the --author etc to the argument slice. Will revisit.